### PR TITLE
[PLT-1107] Add validation for empty row_data for Dataset create_data_rows

### PR DESCRIPTION
**Pull request by vbrodsky**
# Description

During my implementation / refactor of Dataset `create_data_rows`, I somehow dropped validation of empty row_data. This PR restores validation of row_data and throws and error if it is None or an empty string

Fixes the issue discussed here https://labelbox.slack.com/archives/CPVQE6T2T/p1718010997829539

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Document change (fix typo or modifying any markdown files, code comments or anything in the examples folder only)

## All Submissions

- [x] Have you followed the guidelines in our Contributing document?
- [x ] Have you provided a description?
- [x] Are your changes properly formatted?

## Changes to Core Features

- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully run tests with your changes locally?
- [ ] Have you updated any code comments, as applicable?

### CODE REVIEW

**Review by sfendell-labelbox — APPROVED** [review]

File: libs/labelbox/src/labelbox/schema/dataset.py [hunk 1]
```diff
@@ -288,10 +290,9 @@ def create_data_rows(self,
         string_items = [item for item in items if isinstance(item, str)]
         dict_items = [item for item in items if isinstance(item, dict)]
         dict_string_items = []
-
         if len(string_items) > 0:
             dict_string_items = self._build_from_local_paths(string_items)
-        specs = DataRowUpsertItem.build(self.uid,
+        specs = DataRowCreateItem.build(self.uid,
```
Comment (adrian-chang) [inline libs/labelbox/src/labelbox/schema/dataset.py:295]: I thought the point was to maximize use of upsert ... was this on purpose or just for this use case?

Comment (vbrodsky) [reply libs/labelbox/src/labelbox/schema/dataset.py:295]: Yes we are using upsert, and this change just captures difference in data formatting and validation between create and upsert on the sdk side due to a number of edge cases for create

**Comment by vbrodsky** [comment]
Lots of flaky tests, passing locally. Merging PR, will look to address the tests separately
